phoenicis-library/src/main/java/org/phoenicis/library/ShortcutManager.java

```diff
@@ -155,7 +156,46 @@ public void deleteShortcut(ShortcutDTO shortcutDTO) {
     }
 
     public void updateShortcut(ShortcutDTO shortcutDTO) {
+        final String baseName = shortcutDTO.getName();
+        final File shortcutDirectory = new File(this.shortcutDirectory);
+
+        // backup icon if it didn't change (deleteShortcut will delete it -> icon lost after shortcut update)
+        final File iconFile = new File(shortcutDirectory, baseName + ".icon");
+        final File iconBackup = new File(shortcutDirectory, baseName + ".icon_backup");
+
+        final boolean keepIcon = shortcutDTO.getIcon().getPath().equals(iconFile.getPath());
+        if (keepIcon) {
+            try {
+                Files.move(iconFile.toPath(), iconBackup.toPath());
+                shortcutDTO = new ShortcutDTO.Builder(shortcutDTO).withIcon(iconBackup.toURI()).build();
+            } catch (IOException e) {
+                LOGGER.error("Could not backup icon.");
+            }
+        }
+
+        // backup miniature if it didn't change (deleteShortcut will delete it -> miniature lost after shortcut update)
+        final File miniatureFile = new File(shortcutDirectory, baseName + ".miniature");
+        final File miniatureBackup = new File(shortcutDirectory, baseName + ".miniature_backup");
+
+        final boolean keepMiniature = shortcutDTO.getMiniature().getPath().equals(miniatureFile.getPath());
+        if (keepMiniature) {
+            try {
+                Files.move(miniatureFile.toPath(), miniatureBackup.toPath());
+                shortcutDTO = new ShortcutDTO.Builder(shortcutDTO).withMiniature(miniatureBackup.toURI()).build();
+            } catch (IOException e) {
+                LOGGER.error("Could not backup miniature.");
+            }
+        }
+
         deleteShortcut(shortcutDTO);
         createShortcut(shortcutDTO);
+
+        // delete backups
+        if (iconBackup.delete()) {
+            iconBackup.delete();
```

Comment:
I'm not sure I understand these if else blocks, you delete a file after it has been successfully deleted?